### PR TITLE
Add additional docs for rolling updates to migration readme

### PR DESCRIPTION
**Pull request by efritz**
Closes https://github.com/sourcegraph/about/issues/1126.

### CODE REVIEW

File: migrations/README.md [hunk 1]
```diff
@@ -43,12 +43,31 @@ go generate ./migrations/
...
 
 ```
 dev/ci/ci-db-backcompat.sh  # NOTE: this checks out a different git revision, so make sure the work tree is clean before running
 ```
 
+Some migrations are difficult to do in a single step. For instance, renaming a column, table, or view, or
+adding a column with a non-nullable constraint will all break existing code that accesses that table or view.
+In order to do such changes you may need to break your changes into several parts separated by a deployment.
+
+For example, a non-nullable column can be added to an existing table with the following steps:
```
Comment (slimsag) [inline migrations/README.md:59]: These steps seem to go against what is stated below?

> When possible, this same standard should be kept between minor release versions to ensure a smooth upgrade process for private instances

File: migrations/README.md [hunk 2]
```diff
@@ -43,12 +43,31 @@ go generate ./migrations/
 go generate ./internal/db/
 ```
 
-Verify that the migration is backward-compatible:
+Verify that the migration is backward-compatible with the following command. This check will determine the
+version of the code that is running on sourcegraph.com, then run those unit tests against the new schema.
```
Comment (slimsag) [inline migrations/README.md:47]: > This check will determine the version of the code that is running on sourcegraph.com

This is not a true statement, the check doesn't know anything about what is or is not deployed to sourcegraph.com but instead only looks at Git

Comment (efritz) [reply migrations/README.md:47]: https://github.com/sourcegraph/sourcegraph/pull/11814

Comment (efritz) [reply migrations/README.md:47]: See [RFC 181](https://docs.google.com/document/d/1pReWeznvPWQQknXdsK4z7zyhaVzdAcB1FhdDA_b1b24) for context.

File: migrations/README.md (hunk 1 above)
Comment (efritz) [reply migrations/README.md:59]: It's necessary to support rolling updates on dotcom and suggested to support rolling updates for private instances.

**Comment by slimsag** [comment]
I'm gonna need to take a closer look here and have to head out, sorry - happy to defer to others here though.

**Comment by slimsag** [comment]
Merging because I think this is an improvement over the status quo, I'll take ownership of any potential explosions here.

**Pull request merged**